### PR TITLE
feat(automation): add semantic-release & update jenkins

### PR DESCRIPTION
**Pull request by Rifinio**
This PR adds [semantic-release](https://github.com/semantic-release/semantic-release) to automate the package publishing of  `usabilla-react-native`.

### CODE REVIEW

File: usabilla-react-native/package.json [hunk 1]
```diff
@@ -13,11 +13,16 @@
...
     "start": "node node_modules/react-native/local-cli/cli.js start",
-    "test": "jest"
+    "test": "jest",
+    "semantic-release": "semantic-release"
+  },
+  "devDependencies": {
+    "semantic-release": "^12.4.1"
   }
 }
+
```
Comment (tzengerink) [inline usabilla-react-native/package.json]: https://stackoverflow.com/questions/5813311/no-newline-at-end-of-file

File: Jenkinsfile [hunk 2]
```diff
@@ -8,4 +8,13 @@ node('mac') {
     stage('Test') {
         sh 'npm test'
     }
+    if(env.BRANCH_NAME == 'master') {
+        stage('Release') {
+            withCredentials([usernamePassword(credentialsId: '34dc9a59-1608-4470-9860-bd19030cadbb', passwordVariable: 'GH_TOKEN', usernameVariable: 'JENKINS_USER')]) {
+                withCredentials([usernamePassword(credentialsId: 'sdk-npm-token', passwordVariable: 'NPM_TOKEN', usernameVariable: 'NPM_USER')]) {
+                    sh 'npm run semantic-release  || true'
```
Comment (SuzettePaeta) [inline Jenkinsfile]: I think there is an extra space before the ||

**Comment by SuzettePaeta** [comment]
Just out of curiosity how does it work to give the version number?

**Comment by Rifinio** [comment]
@SuzettePaeta it uses normalized conventions for commit messages to determine the next version number.
https://github.com/semantic-release/semantic-release#how-does-it-work

**Pull request merged**